api/client/webclient/webclient.go

```diff
@@ -38,13 +38,22 @@ import (
 	log "github.com/sirupsen/logrus"
 )
 
+type Config struct {
```

Comment:
Nit: Add godocs on the struct and its public fields.

Reply:
Also, add `CheckAndSetDefaults()` method on this struct that makes sure that all mandatory fields are set and sets default values for fields like Context so we don't get panics if they're not set accidentally. Grep the code for other CheckAndSetDefaults methods to see how it's used.